### PR TITLE
Initial grouping of text messages

### PR DESCRIPTION
**Pull request by ljacqu**
@sgdc3 @Twonox,

For #1055 we want to group messages by sections. This is my initial proposition. Please change what you don't like, either by pushing to this branch or submitting a hastebin link

### CODE REVIEW

**Comment by sgdc3** [comment]
Looks fine for me.

**Comment by ljacqu** [comment]
Cool, thanks guys! Can you have a look again @Twonox? ;)

File: src/main/resources/messages/messages_en.yml [hunk 1]
```diff
@@ -58,19 +82,15 @@ email_changed: '&2Email address changed correctly!'
...
-kicked_admin_registered: 'An admin just registered you; please log in again'
 incomplete_email_settings: 'Error: not all required settings are set for sending emails. Please contact an admin.'
+email_already_used: '&4The email address is already being used'
 email_send_failure: 'The email could not be sent. Please contact an administrator.'
-recovery_code_sent: 'A recovery code to reset your password has been sent to your email.'
-recovery_code_incorrect: 'The recovery code is not correct! Use /email recovery [email] to generate a new one'
+show_no_email: '&2You currently don''t have email address associated with this account.'
+add_email: '&3Please add your email to your account with the command "/email add <yourEmail> <confirmEmail>"'
+recovery_email: '&3Forgot your password? Please use the command "/email recovery <yourEmail>"'
+email_already_used: '&4The email address is already being used'
```
Comment (Twonox) [inline src/main/resources/messages/messages_en.yml]: Duplicated (line 86)

File: src/main/resources/messages/messages_en.yml [hunk 2]
```diff
@@ -1,51 +1,75 @@
...
-user_regged: '&cYou already have registered this username!'
-usage_reg: '&cUsage: /register <password> <ConfirmPassword>'
+login_msg: '&cPlease, login with the command "/login <password>"'
+timeout: '&4Login timeout exceeded, you have been kicked from the server, please try again!'
+
+# Errors
+unknown_user: '&cCan''t find the requested user in the database!'
+user_unknown: '&cThis user isn''t registered!'
+denied_command: '&cIn order to use this command you must be authenticated!'
+same_ip_online: 'A player with the same IP is already in game!'
```
Comment (Twonox) [inline src/main/resources/messages/messages_en.yml]: This is error sent when joining. Must be moved.

**Comment by ljacqu** [comment]
Cool, thank you again @Twonox :)

**Comment by Twonox** [comment]
At what moment this message is sent?
`reg_voluntarily: 'You can register yourself to the server with the command "/register <password> <ConfirmPassword>"'`

**Comment by ljacqu** [comment]
@Twonox Definitely, this is on my radar! I want to do the automatic grouping of messages in this pull request, too. Then I can merge as a squash and all the changes here will just be one commit. :wink:

**Comment by Twonox** [comment]
I see, so I did the grouping for french messages for nothing ? xD

**Comment by ljacqu** [comment]
@Twonox In my opinion it was very useful, because you translated the section titles into French, and more importantly this made me realize that however we end up doing the automatic sort, it should be aware that section titles might be translated. :)

**Comment by ljacqu** [comment]
I'm happy that I could merge this now. For the technically involved, a short description:

The `verifyMessages` task's behavior has been changed to also reorder the messages according to the messages_en.yml file and to group them under the same comments.

In particular, this is done the following way:
- Read the messages file into _message file elements_ — this can be a group of comments, or a message entry.
- Do the same for the default messages file (messages_en.yml)

- Go through each _message file element_ of messages_en.yml and get the corresponding element from the local file, if it doesn't exist use the default element. 
   - for a message entry, this is obviously matched via the message key
   - for the comments, there's a counter: "I iterated through the second comments block of messages_en.yml, so I take the second comments block from the local file that I've read (if it exists)"

**Comment by Twonox** [comment]
So, if we modify the order of EN messages, the changes are automatically made to other lang messages ?

**Comment by ljacqu** [comment]
We have a task that we can run that will add the todo comments for missing messages and tags. Now, additionally, it also sorts the messages to be the same as in messages_en.yml.

**Pull request merged**